### ISSUE TITLE
sly/slynk protocol version mismatch

### ISSUE THREAD

**Issue by mfiano** [joaotavora/sly#131]
In either commit 687942c or 6f7d1c5, starting Sly results in this in the modeline on startup:

```
Please answer y or n.  [sly] Versions differ: nil (sly) vs. 1.0.0-beta-2 (slynk). Continue? (y or n)
```
I cannot reproduce this on the previous commit, ae0b383

**Comment by joaotavora** [comment]
Hi Michael,

The TL;DR is that the problem will probably go away if you byte-recompile sly.el or reinstall with any of the latest commits. Simply removing any `sly.elc` from the load path will probably also work (though you should byte-recompile it afterwards). 

I don't know how you installed SLY and how you are conducting these tests. Normally, that info is not relevant, but for this particular problem it is. You see, the only way that elisp variable `sly-protocol-version` can ever be nil is if some non-standard procedures are being used to compile SLY. It could be the (unlikely) case that MELPA (is that how you have it setup) is byte-compiling sly to a non-standard file name.

Can you try the above workarounds and then shed some light on how you are upgrading SLY?

**Comment by mfiano** [comment]
I will try installing it manually when I'm home. I was indeed using melpa
after completely removing all emacs source and byte-compiled files and
directories thereof.

**Comment by mfiano** [comment]
HEAD works fine if I bypass MELPA. Somehow MELPA broke this package since the last 1-2 commits.

**Comment by mindbound** [comment]
Still having this issue when installing Sly in Spacemacs using https://github.com/mfiano/common-lisp-sly.

**Comment by mfiano** [comment]
Yeah, this has been broken in MELPA for a long time. `spacemacs/recompile-elpa` will fix it.

**Comment by mindbound** [comment]
That worked, thanks!

**Comment by stardiviner** [comment]
I got this problem too.
```
[sly] Versions differ: nil (sly) vs. 1.0.0-beta-2 (slynk). Continue? (y or n) y
```

Emacs version: `GNU Emacs 27.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.5) of 2019-02-22`
SLY version: `[sly] SLY 1.0.0-beta-2`
SBCL version: `SBCL 1.4.16`

**Comment by joaotavora** [comment]
@stardiviner how exactly are you 

1. installing SLY? Via git checkout or MELPA?
2. starting SLY? Via `M-x sly` or `M-x sly-connect`?

**Comment by stardiviner** [comment]
1. I install SLY from MELPA
2. I start through `M-x sly`, sometimes sly auto start when in Lisp buffer. Might toggled by `company-sly`.

**Comment by mfiano** [comment]
It's kind of unfortunate that this is still an issue, as I received dozens of questions about this from users installing Sly via MELPA. I wonder what they could be doing to diverge the client/server versions that doesn't occur when installing manually.

**Comment by joaotavora** [comment]
I think SLY might be a victim of MELPA clients pulling some rug under it, i.e. normally the version checking works if `sly.elc` is placed in the same directory as `sly.el`, or something like that.  It might be that some methods of installing via MELPA (there are now several besides `package.el` and `package-install`) might now be violating this expectation.

**Comment by joaotavora** [comment]
Anyway, I'll weigh options, including turning off version checking completely by default

**Comment by joaotavora** [comment]
@mfiano can you confirm if your `sly.el` file is in the same directory as your `sly.elc` file? Are you installing via vanilla `package-install` or using something more fancy such as use-package or "straight" or `el-get` or something?

**Comment by joaotavora** [comment]
@mfiano can you report the value of the Elisp variable `sly-path` immediately before/after the problem happens.  Can you also find the `sly.elc` that you may be loading and attach it here?

**Comment by mfiano** [comment]
@joaotavora I can't reproduce the problem on my current setup. I just had several people ask me about it. I'm using the Doom Emacs distribution, and I think they use a custom package management solution instead of package.el, and I'm not quite sure how it works.

**Comment by stardiviner** [comment]
I use `use-package` and MELPA.

**Comment by stardiviner** [comment]
Seems this issue is fixed, the warning message is gone when today I check again. Thanks for your work!
